### ISSUE TITLE
Improve Performance for TorsionDriveRecord.get_final_molecules()

### ISSUE THREAD

**Issue by yudongqiu** [MolSSI/QCFractal#337]
**Is your feature request related to a problem? Please describe.**
We often need to query individual TorsionDriveRecord for analysis and batches of these for fitting. Currently it takes about 8s for `TorsionDriveRecord.get_final_molecules()` and similar for `TorsionDriveRecord.get_final_results()`. It would be very helpful if these can run faster.

**Describe the solution you'd like**
The `TorsionDriveRecord.get_final_molecules()` and `TorsionDriveRecord.get_final_results()` to run faster. 
Example for `.get_final_molecule()`, we are now querying the final molecule of each geometry optimization task one at a time. [here](https://github.com/MolSSI/QCFractal/blob/994d6a7bc763b2840bba6cae276156ae377fe60d/qcfractal/interface/models/torsiondrive.py#L216)
 It will be much faster if we aggregate all the ids of the final molecules and make one query to get them all.

**Comment by dgasmith** [comment]
I am toying around with the idea of attempting to solve this in a more general manner where we do something like the following:


```
with client.group_query() as query:
    client.query_molecule(id=...)
    client.query_molecule(id=...)
    client.query_molecule(id=...)
```

Where we fuse the intermediate queries, store them in the client, and then repeat the operations to fill in the state. There might be some python magic to do this.

**Comment by yudongqiu** [comment]
Sure I think this could be an interesting general solution for many other usages, however it also requires more careful design to make the general solution correct and convenient. I can imagine different types of query being called and results being sent to other functions inside this block.

Meanwhile I'm also curious if the method I described could be useful for a quick improvement.

**Comment by dgasmith** [comment]
@doaa-altarawy has started a direct (TorsionDrive.id -> FinalMolecule) query for this using SQL joins internally. This will be much faster than any custom query we can write on our end.

**Comment by yudongqiu** [comment]
Sounds great! Looking forward to trying that out.

**Comment by mattwelborn** [comment]
Closed by #393?
